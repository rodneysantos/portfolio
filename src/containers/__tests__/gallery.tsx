import { cleanup, render } from '@testing-library/react';
import React from 'react';
import rendeder from "react-test-renderer";
import Gallery from '../gallery';

afterEach(cleanup);

describe('Gallery component', () => {
  const images = ['image1.jpg', 'image2.jpg', 'image3.jpg'];

  it("renders correctly", () => {
    // act
    const tree = rendeder.create(<Gallery images={images} />);

    // assert
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct number of images', () => {
    // act
    const { getAllByAltText } = render(<Gallery images={images} />);

    // assert
    expect(getAllByAltText('').length).toEqual(images.length);
  });

  it('renders the images with the correct src attributes', () => {
    // act
    const { getAllByAltText } = render(<Gallery images={images} />);

    // assert
    getAllByAltText('').forEach((img, index) => {
      expect((img as HTMLImageElement).src).toContain(images[index]);
    });
  });
});
