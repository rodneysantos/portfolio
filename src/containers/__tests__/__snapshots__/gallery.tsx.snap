// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component renders correctly 1`] = `
<div>
  <img
    alt=""
    onClick={[Function]}
    src="photo1.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="photo2.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="photo3.jpg"
  />
</div>
`;
